setup.py

```diff
@@ -32,21 +35,24 @@
 with open("README.md") as readme_file:
     long_description = readme_file.read()
 
-strict_requirements = {
-    "numba": "numba>=0.4.8",
-    "matplotlib": "matplotlib==3.2.2",
-}
-with open("requirements.txt") as requirements_file:
-    requirements = []
-    # for line in requirements_file:
-    #     # TODO, this should be a proper regex parsing
-    #     requirement, version = line.split("==")
-    #     if requirement not in strict_requirements:
-    #         requirements.append(requirement)
-    #     else:
-    #         requirements.append(strict_requirements[requirement])
-    for line in requirements_file:
-        requirements.append(line)
+
+extra_requirements = {}
+for extra, requirement_file_name in package2install.__requirements__.items():
+    with open(requirement_file_name) as requirements_file:
+        if extra != "":
+            extra_stable = f"{extra}-stable"
+        else:
+            extra_stable = "stable"
+        extra_requirements[extra_stable] = []
+        extra_requirements[extra] = []
+        for line in requirements_file:
+            extra_requirements[extra_stable].append(line)
+            requirement, *comparison = re.split("[><=~!]", line)
+            requirement == requirement.strip()
+            extra_requirements[extra].append(requirement)
+
+requirements = extra_requirements.pop("")
+
 
 def is_tool(name):
     """Check whether `name` is on PATH and marked as executable."""
```

Comment:
Did you check if this actually works? I would expect this only to work if you install from source. When creating a wheel for e.g. python, this is probably not going to work, since this is checked on the machine that CREATES the wheel and not in the machine that USES the wheel